### PR TITLE
1431: Add app build config app

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Added app build config for koblenz

### Proposed changes
- added icons and images
- added build config
- added run configs for intellij
- adjust build gradle
- add koblenz german texts
- added appstore and development profiles for koblenz zu apple and credentials repository
- added xcode schemes and assets

Still open tasks due to currently still missing data will be done in https://github.com/digitalfabrik/entitlementcard/issues/1653
Known issue, will be fixed in separate task: https://github.com/digitalfabrik/entitlementcard/issues/1686

### Testing
Start the app with koblenz flavour
check if deep link files are created properly
check on android and ios
check if correct intro screens and images / icons are displayed
check if correct publisher text is displayed
check if map is centered correctly to koblenz

### Resolved issues

Fixes: #1431

### CODE REVIEW

File: frontend/assets/koblenz/l10n/override_de.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+  "identification": {
+    "activateCurrentDeviceDescription": "Ihr Pass wurde bereits auf einem anderen Gerät aktiviert. Wenn Sie fortfahren, wird Ihr Pass automatisch auf dem anderen Gerät deaktiviert, falls er dort noch aktiv ist.",
+    "activateCurrentDeviceTitle": "Pass auf diesem Gerät aktivieren?",
+    "activateDescription": "Sie haben den Koblenz-Pass bereits beantragt und einen Aktivierungscode erhalten? Scannen Sie den Code hier ein.",
+    "activateTitle": "Pass aktivieren",
+    "applyDescription": "Sie haben noch keinen Koblenz-Pass? Hier können Sie Ihren Koblenz-Pass beantragen.",
```
Comment (seluianova) [inline frontend/assets/koblenz/l10n/override_de.json:7]: as I understood, it should be KoblenzPass everywhere

Comment (f1sh1918) [reply frontend/assets/koblenz/l10n/override_de.json:7]: Yes

File: frontend/build-configs/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export {default as buildConfigNuernberg} from './nuernberg'
-export {default as buildConfigBayern} from './bayern'
-export type {DeeplLinkingConfig, BuildConfigType} from './types'
-export * from './constants'
\ No newline at end of file
+export { default as buildConfigNuernberg } from "./nuernberg"
+export { default as buildConfigBayern } from "./bayern"
+export { default as buildConfigKoblenz } from "./koblenz"
+export type { DeeplLinkingConfig, BuildConfigType } from "./types"
+export * from "./constants"
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply frontend/build-configs/index.ts:5]: Same for other files. Shouldn't this be caught by the CI pipeline/prettier?

**Comment by f1sh1918** [comment]
Tested on pixel 7 and found one issue. There is an overflow of the id card widget. Since we are not releasing this it can be fixed later but we should keep this in mind

I also get an error in the console when starting the app
```js
flutter: Environment: local
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Unable to load asset: "assets/koblenz/l10n/override_en.json".
```

image

Could you please add in `getBuildConfig` LN25 and remove TODO
```
return buildConfigKoblenz
```
I don't think we need a separate task for this one liner.
This also fixes deeplinking on ios for me
image

File: frontend/build-configs/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ztefanie) [reply frontend/build-configs/index.ts:5]: It is quite inconsistent currently, as these ts files are unter /frontend and prettier is only in /administration. See e.g. build-config/bayern/index has also double quotes. I can adjust this, but i would prefer to do it in a separate PR, so this PR does not get too big.

File: frontend/build-configs/koblenz/index.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        },
+        ios: {
+            appleAppSiteAssociationAppId: `7272KE28TJ.${IOS_BUNDLE_IDENTIFIER}`,
+            path: `/${ACTIVATION_PATH}/*`,
+            pathComment: `Matches any URL with a path that starts with /${ACTIVATION_PATH}/.`,
+        },
+    },
+}
+
+let koblenz: BuildConfigType = {
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply frontend/build-configs/koblenz/index.ts:97]: hmh, you are right, it is let for the others configs too. I will adjust it everywhere

**Review by f1sh1918 — APPROVED** [review]
Very good work 👍 
Tested on ios and android.
Would be nice if you can adjust this one line i mentioned in my comment before
Then its good to go

File: frontend/build-configs/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply frontend/build-configs/index.ts:5]: Oh okay, thanks. Yes, then a separate PR makes sense.

**Comment by ztefanie** [comment]
> Tested on android, works as expected. What do you mean by "check that deeplink files are created"?

When building administration in the build folder there are the files apple-app-site-association and assentlinks.json for each project

**Comment by ztefanie** [comment]
> Please add issues for the two minor things i mentioned there
They are already in the issue in the description of the PR, i added the console output for the missing english translations

**Comment by f1sh1918** [comment]
> > Please add issues for the two minor things i mentioned there
> 
> They are already in the issue in the description of the PR, i added the console output for the missing english translations

okay i didn't find adjust theme color, so i added this to #1653

**Pull request merged**